### ISSUE TITLE
Should we teach getwd()?

### ISSUE THREAD

**Issue by monicathieu** [cu-psych-computing/cu-psych-comp-tutorial#39]
In the [data cleaning lesson plan](https://github.com/cu-psych-r-users/cu-psych-r-tutorial/blob/c65de834382e728935a216694afe846154409298/2_dataCleaning/DataCleaning_Code.Rmd#L66), we reference `getwd()` to show the working directory and `list.files()` to show files in said directory, as a precursor to reading in a CSV file. I'm not convinced that this is cleanest practice, as it gets a little dangerously close to `setwd()`. Flagging here for discussion.

**Comment by emilynakka** [comment]
Rather than focusing on `getwd()` and `list.files()`, why don't we introduce early on how to access files from a data folder in the same project directory? Something like... `read.csv("data/mydata.csv")` or `read.csv("../data/mydata.csv")` if the analysis file is already in a separate subdirectory?

As an added benefit, I think this could help students institute valuable organizational practices early on.

That said, if you guys think this is too complex at this stage in the workshop, completely understand!

**Comment by monicathieu** [comment]
If we can add something to get around the occasional issue that .Rmd files tend to assume a working directory of "wherever the file is saved", which sometimes isn't the same as the project directory (like if you have Rmds in a subfolder), this kind of setup would be good! The `here` package ([Jenny Bryan recommends it](https://github.com/jennybc/here_here)) might be an option that we can bring into the workflow to make darn sure everyone's Rmd working directory is the same as their project directory.

**Comment by emilynakka** [comment]
Definitely! As a package-free alternative, we could also suggest that in a project directory, students *always* create a few subfolders (e.g., /data, /analysis, /output) and save all data to /data and all .Rmds to /analysis. Then, they could stick `knitr::opts_knit$set(root.dir = normalizePath('../'))` in the setup chunk of every .Rmd they create. I really would have appreciated this kind of directory-organization guidance when I was learning R, but also, I am weird and maybe this is too complex. I leave it to you guys to decide which option is right for this tutorial. :)

**Comment by emilynakka** [comment]
(Also, `here` looks like a cool package—thanks for sharing!)

**Comment by monicathieu** [comment]
Decided to go 100% with `here()` and replaced the use of `getwd()` with another `here()` call (should work, since the pkg has been loaded at that point). Leaves the utility of showing the current working directory, and ameliorates the confusion introduced when Rmds treat _their containing folder_ as the working directory when knitting, not the project home. (I had many of those path errors a year ago...) Should have been addressed by #65 that I made just now.
